### ISSUE TITLE
LSP: don't send didChange before didOpen

### ISSUE THREAD

**Issue by breja** [veonim/veonim#475]
this was fixed before and you broke it again. you should feel ashamed.
![frozendissapointed](https://user-images.githubusercontent.com/2581815/45060202-fc42e880-b06c-11e8-9f02-0baf8e94d492.gif)
![you-trusted-me](https://user-images.githubusercontent.com/2581815/45060147-bb4ad400-b06c-11e8-86a2-87e5354ab4f0.gif)

**Comment by breja** [comment]
so the problem is in the buffer calls until server starts workflow. this is async of course, and async makes everything messy

here's the flow that causes this to go out of order
```
didOpen
queue await didOpen.server
didChange
queue await didChange.server
resolve didChange.server -> no server -> buffer call
lang server starts
empty buffer -> send didChange
resolve await didOpen.server
send didOpen
```

The only clean way to solve this is to keep track of the didOpen/didChange state inside veonim. Then we need to queue up the didChange/didOpen calls and send them in order when the server starts.

We need to keep in mind that not all servers support didOpen (is that true? verify), so we should check supported server capabilities before queuing didChanges behind didOpen.

**Comment by breja** [comment]
fixed with TDM #503
